frontend/lib/resolution/prims.cpp

```diff
@@ -511,6 +526,255 @@ static QualifiedType primObjectToInt(Context* context, const CallInfo& ci) {
+  if (ci.numActuals() < 1) return {};
+  bool eval = false;
+  if (auto t = ci.actual(0).type().type()) {
+    if (auto mt = t->toClassType()) eval = mt->decorator().isNonNilable();
+  }
+  return makeParamBool(context, eval);
+}
+
+static QualifiedType primIsRecordType(Context* context, const CallInfo& ci) {
+  if (ci.numActuals() < 1) return {};
+  bool eval = false;
+  if (auto t = ci.actual(0).type().type()) eval = t->isUserRecordType();
+  return makeParamBool(context, eval);
+}
+
+static QualifiedType primIsFcfType(Context* context, const CallInfo& ci) {
+  CHPL_UNIMPL("PRIM_IS_FCF_TYPE");
+  return {};
+}
+
+static QualifiedType primIsUnionType(Context* context, const CallInfo& ci) {
+  if (ci.numActuals() < 1) return {};
+  bool eval = false;
+  if (auto t = ci.actual(0).type().type()) eval = t->isUnionType();
+  return makeParamBool(context, eval);
+}
+
+static QualifiedType primIsExternType(Context* context, const CallInfo& ci) {
+  CHPL_UNIMPL("PRIM_IS_EXTERN_TYPE");
+  return {};
+}
+
+static QualifiedType
+primIsExternUnionType(Context* context, const CallInfo& ci) {
+  auto qt1 = primIsExternType(context, ci);
+  auto qt2 = primIsUnionType(context, ci);
+  const bool eval = qt1.isParamTrue() && qt2.isParamTrue();
+  return makeParamBool(context, eval);
+}
+
+static QualifiedType
+primIsAtomicType(Context* context, const CallInfo& ci) {
+  if (ci.numActuals() < 1) return {};
+  bool eval = false;
+  if (auto t = ci.actual(0).type().type()) {
+    eval = t->hasPragma(context, PRAGMA_ATOMIC_TYPE);
+  }
+  return makeParamBool(context, eval);
+}
+
+static QualifiedType
+primIsRefIterType(Context* context, const CallInfo& ci) {
+  CHPL_UNIMPL("PRIM_IS_REF_ITER_TYPE");
+  return {};
+}
+
+static QualifiedType
+primIsBorrowedClassType(Context* context, const CallInfo& ci) {
+  if (ci.numActuals() < 1) return {};
+  bool eval = false;
+  if (auto t = ci.actual(0).type().type()) {
+    if (auto mt = t->toClassType()) eval = mt->decorator().isBorrowed();
+  }
+  return makeParamBool(context, eval);
+}
+
+static QualifiedType
+primIsAbsEnumType(Context* context, const CallInfo& ci) {
+  if (ci.numActuals() < 1) return {};
+  bool eval = false;
+  if (auto t = ci.actual(0).type().type()) {
+    if (auto et = t->toEnumType()) eval = et->isAbstract();
+  }
+  return makeParamBool(context, eval);
+}
+
+static QualifiedType
+primIsPod(Context* context, const CallInfo& ci) {
+  CHPL_UNIMPL("PRIM_IS_POD");
+  return {};
+}
+
+static QualifiedType
+primIsCoercible(Context* context, const CallInfo& ci) {
+  if (ci.numActuals() < 2) return {};
+
+  // In the old compiler this primitive only considers 'type'.
+  const auto tpFrom = ci.actual(0).type().type();
+  const auto tpTo = ci.actual(1).type().type();
+  const auto qtFrom = QualifiedType(QualifiedType::TYPE, tpFrom);
+  const auto qtTo = QualifiedType(QualifiedType::TYPE, tpTo);
+
+  const auto canPass = CanPassResult::canPass(context, qtFrom, qtTo);
```

Comment:
Note that you only had to add `const` to all the fields of `canPass` because you explicitly specified `const` here (clearly not done anywhere else in Dyno, or the other people would've been hit by it 😄 ). IMO, you could write this without `const` to no real downside at all.

Reply:
I think it would work either way, it's just a style thing. Some of the new code Anna has added also uses `const`. I'm ok with using it or not, I think it just documents intent (even if it's a bit wordier).